flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java

```diff
@@ -34,29 +58,131 @@ private SqlCommandParser() {
 		// private
 	}
 
-	public static Optional<SqlCommandCall> parse(String stmt) {
+	public static Optional<SqlCommandCall> parse(Function<String, List<Operation>> sqlParserFunction, String stmt) {
 		// normalize
 		stmt = stmt.trim();
```

Comment:
How about we just split 2 methods `parseByExecutor` and `parseByCli`, parse a function here seems weird, because the first operand of the `sqlParserFunction ` is just the `stmt`.

Reply:
for the `SqlCommandParser` users, they do not need to know how to parse a statement, so I think we should only provide one entry point method. I can extract the regex matching part into a method. for `sqlParserFunction`, how about pass it into `SqlCommandParser` through constructor, that means we should make `parse` method as a non-static method ?

Reply:
How about just add method `#parse(Parser, String)`, and add a new interface `Executor#getParser`.